src/hpc/tls.h

```diff
@@ -55,19 +55,40 @@ typedef struct ThreadLocalStorage
   UInt LocksContended;
 } ThreadLocalStorage;
 
-#ifdef HAVE_NATIVE_TLS
+#ifdef USE_NATIVE_TLS
 extern __thread ThreadLocalStorage *TLSInstance;
 #endif
 
+#ifdef USE_PTHREAD_TLS
+
+#include <pthread.h>
+
+#ifdef __GNUC__
+// This trick allows gcc/clang to hoist calls to pthread_getspecific()
+// out of loops and perform constant subexpression elimination on
+// expressions containing TLS values.
+__attribute__((pure)) void * pthread_getspecific(pthread_key_t);
+#endif
+
+static pthread_key_t TLSKeyCopy;
+pthread_key_t GetTLSKey(void);
+
+__attribute__((constructor))
```

Comment:
Why is this attribute here (needed) ? Let's either add a comment explaining that, or get rid of it.

Reply:
Explanation added. Briefly, the constructor attribute results in execution of the code when the program is started/library is loaded in order to initialize the static variable.